lib/manager/flux/extract.ts

```diff
@@ -0,0 +1,123 @@
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type {
+  FluxManifest,
+  FluxResource,
+  HelmRelease,
+  HelmRepository,
+} from './types';
+
+function readManifest(content: string): FluxManifest | null {
+  const manifest: FluxManifest = { releases: [], repositories: [] };
+  let resources: FluxResource[];
+  try {
+    resources = loadAll(content, null, { json: true }) as FluxResource[];
+  } catch (err) {
+    logger.debug({ err }, 'Failed to parse Flux manifest');
+    return null;
+  }
+
+  // It's possible there are other non-Flux HelmRelease/HelmRepository CRs out there, so we filter based on apiVersion.
+  for (const resource of resources) {
+    switch (resource?.kind) {
+      case 'HelmRelease':
+        if (
+          resource.apiVersion?.startsWith('helm.toolkit.fluxcd.io/') &&
+          resource.spec?.chart?.spec?.chart
+        ) {
+          manifest.releases.push(resource);
+        }
+        break;
+      case 'HelmRepository':
+        if (
+          resource.apiVersion?.startsWith('source.toolkit.fluxcd.io/') &&
+          resource.metadata?.name &&
+          resource.metadata.namespace &&
+          resource.spec?.url
+        ) {
+          manifest.repositories.push(resource);
+        }
+        break;
+    }
+  }
+
+  return manifest;
+}
+
+function resolveReleases(
+  releases: HelmRelease[],
+  repositories: HelmRepository[]
+): PackageDependency[] {
+  return releases.map((release) => {
+    const res: PackageDependency = {
+      depName: release.spec.chart.spec.chart,
+      currentValue: release.spec.chart.spec.version,
+    };
+
+    const matchingRepositories = repositories.filter(
+      (rep) =>
+        rep.kind === release.spec.chart.spec.sourceRef?.kind &&
+        rep.metadata.name === release.spec.chart.spec.sourceRef.name &&
+        rep.metadata.namespace ===
+          (release.spec.chart.spec.sourceRef.namespace ||
+            release.metadata?.namespace)
+    );
+    if (matchingRepositories.length) {
+      res.registryUrls = matchingRepositories.map((repo) => repo.spec.url);
+    } else {
+      res.skipReason = SkipReason.UnknownRegistry;
+    }
+
+    return res;
+  });
+}
+
+export function extractPackageFile(content: string): PackageFile | null {
+  const manifest = readManifest(content);
+  if (!manifest) {
+    return null;
+  }
+  const deps = resolveReleases(manifest.releases, manifest.repositories);
+  return deps.length ? { deps: deps, datasource: HelmDatasource.id } : null;
+}
+
+export async function extractAllPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  const releases = new Map<string, HelmRelease[]>();
+  const repositories: HelmRepository[] = [];
+  const results: PackageFile[] = [];
+
+  for (const file of packageFiles) {
+    const content = await readLocalFile(file, 'utf8');
+    if (!content) {
+      logger.debug({ file }, 'Empty or non existent Flux manifest');
```

Comment:
```suggestion
      logger.debug({ file }, 'Empty Flux manifest');
```
File exists at that stage.

Reply:
Removed this whole `if` block - there's no real need to log empty files or treat them as a special case.